offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/projects/ProjectMainConnector.groovy

```diff
@@ -78,13 +81,37 @@ class ProjectMainConnector implements CreateProjectDataSource, CreateProjectSpac
     //projectDbConnector.fetchProjects() might be used at some point to fetch more metadata
       
     openbisProjects = []
-    for(Project openbisProject : openbisClient.listProjects()) {
-      String space = openbisProject.getSpace().getCode()
-      String code = openbisProject.getCode()
-      openbisProjects.add(new ProjectIdentifier(space, code))
+    for(ch.ethz.sis.openbis.generic.asapi.v3.dto.project.Project openbisProject : openbisClient.listProjects()) {
+      try {
+        ProjectSpace space = new ProjectSpace(openbisProject.getSpace().getCode())
+        ProjectCode code = new ProjectCode(openbisProject.getCode())
```

Comment:
are these optionals? I would opt for testing if the respective space and code are there and throw specific errors rather than do it like this

Reply:
no, they are not optionals

Reply:
POJOs